selfdrive/car/nissan/values.py

```diff
@@ -20,29 +19,52 @@ def __init__(self, CP):
     pass
 
 
-class CAR(StrEnum):
-  XTRAIL = "NISSAN X-TRAIL 2017"
-  LEAF = "NISSAN LEAF 2018"
-  # Leaf with ADAS ECU found behind instrument cluster instead of glovebox
-  # Currently the only known difference between them is the inverted seatbelt signal.
-  LEAF_IC = "NISSAN LEAF 2018 Instrument Cluster"
-  ROGUE = "NISSAN ROGUE 2019"
-  ALTIMA = "NISSAN ALTIMA 2020"
-
-
 @dataclass
 class NissanCarInfo(CarInfo):
   package: str = "ProPILOT Assist"
   car_parts: CarParts = field(default_factory=CarParts.common([CarHarness.nissan_a]))
 
 
-CAR_INFO: dict[str, NissanCarInfo | list[NissanCarInfo] | None] = {
-  CAR.XTRAIL: NissanCarInfo("Nissan X-Trail 2017"),
-  CAR.LEAF: NissanCarInfo("Nissan Leaf 2018-23", video_link="https://youtu.be/vaMbtAh_0cY"),
-  CAR.LEAF_IC: None,  # same platforms
-  CAR.ROGUE: NissanCarInfo("Nissan Rogue 2018-20"),
-  CAR.ALTIMA: NissanCarInfo("Nissan Altima 2019-20", car_parts=CarParts.common([CarHarness.nissan_b])),
-}
+@dataclass
+class NissanCarSpecs(CarSpecs):
+  centerToFrontRatio: float = field(default=0.44)
+  steerRatio: float = field(default=17.)
+
+
+@dataclass
+class NissanPlaformConfig(PlatformConfig):
+  dbc_dict: DbcDict = field(default_factory=lambda: dbc_dict('nissan_x_trail_2017_generated', None))
+
+
+class CAR(Platforms):
+  XTRAIL = NissanPlaformConfig(
+    "NISSAN X-TRAIL 2017",
+    NissanCarInfo("Nissan X-Trail 2017"),
+    specs=NissanCarSpecs(mass=1610, wheelbase=2.705)
+  )
+  LEAF = NissanPlaformConfig(
+    "NISSAN LEAF 2018",
+    NissanCarInfo("Nissan Leaf 2018-23", video_link="https://youtu.be/vaMbtAh_0cY"),
+    dbc_dict=dbc_dict('nissan_leaf_2018_generated', None),
+    specs=NissanCarSpecs(mass=1610, wheelbase=2.705)
+  )
+  # Leaf with ADAS ECU found behind instrument cluster instead of glovebox
+  # Currently the only known difference between them is the inverted seatbelt signal.
+  LEAF_IC = LEAF.override(platform_str="NISSAN LEAF 2018 Instrument Cluster", car_info=None)
```

Comment:
should we just use replace here directly?

Reply:
yea frozen makes sense, and `replace` also seems fine

Reply:
hmm but looks like you can't subclass a frozen dataclass...

Reply:
https://github.com/commaai/openpilot/pull/31605

Reply:
hmm, replace is currently used by `str` since platformconfig currently subclasses `str`